source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -284,35 +288,23 @@ void Filter::populateDescriptors(std::vector<RateLimit::LocalDescriptor>& descri
 void Filter::populateDescriptors(const Router::RateLimitPolicy& rate_limit_policy,
                                  std::vector<RateLimit::LocalDescriptor>& descriptors,
                                  Http::RequestHeaderMap& headers) {
-  const auto* config = getConfig();
   for (const Router::RateLimitPolicyEntry& rate_limit :
-       rate_limit_policy.getApplicableRateLimit(config->stage())) {
+       rate_limit_policy.getApplicableRateLimit(used_config_->stage())) {
     const std::string& disable_key = rate_limit.disableKey();
 
     if (!disable_key.empty()) {
       continue;
     }
-    rate_limit.populateLocalDescriptors(descriptors, config->localInfo().clusterName(), headers,
-                                        decoder_callbacks_->streamInfo());
+    rate_limit.populateLocalDescriptors(descriptors, used_config_->localInfo().clusterName(),
+                                        headers, decoder_callbacks_->streamInfo());
   }
 }
 
-const FilterConfig* Filter::getConfig() const {
-  const auto* config =
-      Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfig>(decoder_callbacks_);
-  if (config) {
-    return config;
-  }
-
-  return config_.get();
-}
```

Comment:
we needn't search and get route config repeatedly. For most filters, in most cases, only once searching is necessary.
So, we removed the method incase someone abuses the method in the future. Calling `getConfig()` is too easy and  may lets the developers to use it without a deep thinking.